src/java/main/org/apache/zookeeper/server/upgrade/UpgradeMain.java

```diff
@@ -113,16 +113,18 @@ private void createAllDirs() throws IOException {
      * @throws IOException
      */
     void copyFiles(File srcDir, File dstDir, String filter) throws IOException {
-        File[] list = srcDir.listFiles();
-        for (File file: list) {
-            String name = file.getName();
-            if (name.startsWith(filter)) {
-                // we need to copy this file
-                File dest = new File(dstDir, name);
-                LOG.info("Renaming " + file + " to " + dest);
-                if (!file.renameTo(dest)) {
-                    throw new IOException("Unable to rename " 
-                            + file + " to " +  dest);
+        File[] list;
+        if ((list = srcDir.listFiles()) != null) {
```

Comment:
Please keep var assignment `(list = srcDir.listFiles())` along with the object reference.

`File[] list = srcDir.listFiles();`